File: 19주차/NH/Swap Nodes in Pairs.js
function swapPairs(head) {
    if(!head || !head.next) return head;
   let v1 = head
   let v2 = head.next
   let v3 = v2.next;
   v2.next = v1;
   v1.next = swapPairs(v3);
   return v2;
};

File: 19주차/SG/Swap Nodes in Pairs.cpp
#include<iostream>
using namespace std;

// Definition for singly-linked list.
struct ListNode {
      int val;
      ListNode *next;
      ListNode() : val(0), next(nullptr) {}
      ListNode(int x) : val(x), next(nullptr) {}
      ListNode(int x, ListNode *next) : val(x), next(next) {}
 };
 
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        int temp;
        ListNode *ptr = head;
        
        while(head != NULL){
            
            if(head->next != NULL){
                temp = (head->next)->val;
                (head->next)->val = head->val;
                head->val = temp;   
            }
            else
                break;
            
            head = head->next->next;
        }
        
        return ptr;
    }
};